### PR TITLE
refactor: treat illegal intrinsic as soft failure

### PR DESCRIPTION
**Pull request by jaschdoc**
@magnus-madsen this change treats illegal intrinsics as soft failures so we can get started on the vscode end of the pipeline

Related to https://github.com/flix/flix/issues/5026

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Weeder.scala [hunk 1]
```diff
@@ -748,7 +748,7 @@ object Weeder {
 
           case ("SCOPE_EXIT", e1 :: e2 :: Nil) => WeededAst.Expression.ScopeExit(e1, e2, loc).toSuccess
 
-          case _ => WeederError.IllegalIntrinsic(loc).toFailure
+          case _ => Validation.SoftFailure(WeededAst.Expression.Wild(loc), LazyList(IllegalIntrinsic(loc)))
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Weeder.scala:751]: Why does it become WeededAst.Expression.Wild? Should it not be Expression.Error?

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/Weeder.scala:751]: Oh yes of course! I took too much time off :D

**Comment by magnus-madsen** [comment]
Seems it continues until the backend, but the Flix check should stop earlier.

**Comment by magnus-madsen** [comment]
I am working on #5360

**Comment by magnus-madsen** [comment]
In this call:

```
WeededAst.Expression.Error(IllegalIntrinsic(loc)).toSoftFailure
```

we are not actually putting the error into the list. So perhaps we need to change the signature of `toSoftFailure`.

**Comment by magnus-madsen** [comment]
But we kind of need two versions: One that just soft fails and one that soft fails and puts the error into the applicative functor... Hm

**Comment by magnus-madsen** [comment]
I fixed it in #5361, but please add the (missing) test case and continue with the pattern I have in Weeder:

```
            val err = WeederError.IllegalIntrinsic(loc)
            WeededAst.Expression.Error(err).toSoftFailure(err)
```

Note I call it `err` because we use `e` for expressions.

**Comment by jaschdoc** [comment]
I'm not sure what test case you are referring to? I added https://github.com/flix/flix/blob/40a58e4d150f867d0a0a4509323a0490d3e7d9fe/main/test/ca/uwaterloo/flix/language/phase/TestWeeder.scala#L751

**Comment by jaschdoc** [comment]
Should we perhaps do that in a separate PR?

**Comment by magnus-madsen** [comment]
Yep

**Pull request merged**